internal/backends/postgresql/query_test.go

```diff
@@ -0,0 +1,255 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package postgresql
+
+import (
+	"math"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/FerretDB/FerretDB/internal/backends/postgresql/metadata"
+	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/must"
+)
+
+func TestPrepareWhereClause(t *testing.T) {
+	t.Parallel()
+	objectID := types.ObjectID{0x62, 0x56, 0xc5, 0xba, 0x0b, 0xad, 0xc0, 0xff, 0xee, 0xff, 0xff, 0xff}
+
+	// WHERE clauses occurring frequently in tests
+	whereContain := " WHERE _jsonb->$1 @> $2"
+	whereGt := " WHERE _jsonb->$1 > $2"
+	whereNotEq := ` WHERE NOT ( _jsonb ? $1 AND _jsonb->$1 @> $2 AND _jsonb->'$s'->'p'->$1->'t' = `
+
+	for name, tc := range map[string]struct {
+		filter   *types.Document
+		expected string
+		skip     string
+		args     []any // if empty, check is disabled
```

Comment:
Why do we disable check for args? I know we did it this way previously but just commenting it makes it difficult for readers.

Currently, test cases are a bit difficult to read because expected where clause such as `whereContain` is extracted top of the function, so each test case I go up to the top of the test to read what is expected.

Seeing something like below to me is easier to tell input `filter`, `expected` where clause and `args`. :hugs: 
```
		"EqObjectID": {
			filter: must.NotFail(types.NewDocument(
				"v", must.NotFail(types.NewDocument("$eq", objectID)),
			)),
			expected: " WHERE _jsonb->$1 @> $2",
			args: []any{`v`, `"6256c5baadc0ffeeffffff"`},
		},
```

Reply:
It's an optional comment, because after all we test what we want to test.

Reply:
We use args mainly to validate that we send proper arguments in edge case scenarios. It's optional as adding expected values such as 42 or "foo" for every test doesn't make a lot of sense, but it takes more lines of code, and creating new tests is just more difficult.

Reply:
We always copy and paste for test cases, so I don't think creating new test would be harder :see_no_evil: